test/unit_test/rag/test_naive_merge.py

```
@@ -149,23 +149,24 @@ def test_empty_delimiter_falls_back_to_token_size_merge():
     # UNDER_CAP (strict): content chunks never overflow chunk_token_num, so the
     # overlap-prefix budget check is the only thing under test here.
     chunks = _nonempty(naive_merge(sentences, chunk_token_num=50, delimiter=DEFAULT_DELIMITER, overlapped_percent=20, strategy=MergeStrategy.UNDER_CAP))
     assert len(chunks) > 1
-    # Each content chunk stays within the budget. Sentences are 10 tokens, the
-    # budget is 50, so a 5-sentence chunk is exactly 50; a 10-token overlap
-    # prefix (20% of 50) would push it to 60 and is therefore dropped at the
-    # boundary rather than letting the chunk overshoot.
-    assert all(_tok(c) <= 50 for c in chunks)
+    # The overlap prefix is always present at every boundary: each chunk (after
+    # the first) starts with the tail of the previous chunk.
+    for prev, cur in zip(chunks, chunks[1:]):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)